### PR TITLE
Improve solutions/analysis for Mountain View and Rental Service

### PR DESCRIPTION
**Pull request by shaubby**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.

### CODE REVIEW

File: solutions/silver/usaco-787.mdx [hunk 1]
```diff
@@ -2,11 +2,33 @@
...
+
+Remember, at any time Farmer John should either milk the highest-producing cow or rent the lowest-producing cow based on whichever one makes the most money.
+
+</Spoiler>
+
+<Spoiler title="Hint 2">
+
+Farmer John should always sell milk to the store which offers the most money, and rent to the farmer which offers the most money.
+
+</Spoiler>
```
Comment (SansPapyrus683) [inline solutions/silver/usaco-787.mdx:18]: these two hints seem to say the same thing?

File: solutions/silver/usaco-896.mdx [hunk 2]
```diff
@@ -6,6 +6,35 @@ author: Jesse Choe, Sofia Yang, Juheon Rhee
...
+
+![](usaco-896/mountains2.png)
+
+In this case, $\mathcal{j}$, $\mathcal{i}_{x}+{i}_{y}\le{j}_{x}+{j}_{y}$, but $\mathcal{i}_{x}-{i}_{y}<{j}_{x}-{j}_{y}$, meaning  that the  rightmost end of the mountain $\mathcal{i}$ is to the left of the rightmost end of mountain $\mathcal{j}$, but the leftmost end of mountain $\mathcal{i}$ is not to the right of the leftmost end of mountain $\mathcal{j}$, meaning it is not contained within mountain j.
+
+This same reasoning can also work if the mountains overlap, as shown below:
+
+![](usaco-896/mountains3.png)
+
+Using this information, we can loop through every pair of mountains and check if one is occluded by another, and complete this problem in $\mathcal{O}(n^2)$ time complexity, but that may be too slow and not complete every testcase (our Java implementation of this method completed 10/11 testcases), so we can use an alternative approach.
```
Comment (SansPapyrus683) [inline solutions/silver/usaco-896.mdx:34]: just say "naively going through all pairs is too slow lol"

File: solutions/silver/usaco-896.mdx [hunk 3]
```diff
@@ -7,6 +7,36 @@ author: Jesse Choe, Sofia Yang, Juheon Rhee
...
+In *this* case, $\mathcal{j}$, $\mathcal{i}_{x}+{i}_{y}\le{j}_{x}+{j}_{y}$, but $\mathcal{i}_{x}-{i}_{y}<{j}_{x}-{j}_{y}$, meaning that the right end of the mountain $\mathcal{i}$ is to the left of the rightmost end of mountain $\mathcal{j}$, but the left end of mountain $\mathcal{i}$ is not to the right of the left end of mountain $\mathcal{j}$, meaning it is not contained within mountain j.
+
+This same reasoning can also work if the mountains overlap, as shown below:
+
+![](usaco-896/mountains3.png)
+
+Using this information, we can loop through every pair of mountains and check if one is occluded by another, and complete this problem in $\mathcal{O}(n^2)$ time complexity, but that may be too slow and not complete every testcase (our Java implementation of this method completed 10/11 testcases), so we can use an alternative approach.
+
+If we sort the mountains by their left end.
+Then we can loop through the list and keep track of the highest right end.
```
Comment (SansPapyrus683) [inline solutions/silver/usaco-896.mdx:37]:
```suggestion
Let's sort the mountains by their left end first and then iterate through all of them, keeping track of the highest right end we've seen.
```

File: solutions/silver/usaco-896.mdx [hunk 4]
```diff
@@ -7,6 +7,36 @@ author: Jesse Choe, Sofia Yang, Juheon Rhee
...
+
+In this case, In this case, $\mathcal{j}$, $\mathcal{i}_{x}+{i}_{y}\le{j}_{x}+{j}_{y}$ and $\mathcal{i}_{x}-{i}_{y}\ge{j}_{x}-{j}_{y}$, meaning that the left and right ends of the mountain $\mathcal{i}$ are contained within mountain $\mathcal{j}$.
+
+![](usaco-896/mountains2.png)
+
+In *this* case, $\mathcal{j}$, $\mathcal{i}_{x}+{i}_{y}\le{j}_{x}+{j}_{y}$, but $\mathcal{i}_{x}-{i}_{y}<{j}_{x}-{j}_{y}$, meaning that the right end of the mountain $\mathcal{i}$ is to the left of the rightmost end of mountain $\mathcal{j}$, but the left end of mountain $\mathcal{i}$ is not to the right of the left end of mountain $\mathcal{j}$, meaning it is not contained within mountain j.
+
+This same reasoning can also work if the mountains overlap, as shown below:
+
+![](usaco-896/mountains3.png)
```
Comment (SansPapyrus683) [inline solutions/silver/usaco-896.mdx:32]: ok i'm all for handholding but this seems a bit _too_ obvious

i don't think it should take this many lines to say "a mountain is hidden if it's 100% in the interval of another"

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/silver/usaco-787.mdx [hunk 5]
```diff
@@ -2,11 +2,33 @@
...
+
+<Spoiler title="Solution">
+
 [Official Analysis (Java)](http://www.usaco.org/current/data/sol_rental_silver_jan18.html)
 
+## Explanation
+
+To decide whether or not to milk or rent a cow, we should always compare the highest-producing cow with the highest offering store, and the highest offering farmer.
+
+To do this, we can make 3 lists of stores, farmers, and cows, and sort the stores and farmers by the prices offered, and sort the cows by their milk-production.
```
Comment (ryanchou-dev) [inline solutions/silver/usaco-787.mdx:28]: break this sentence up

File: solutions/silver/usaco-787.mdx [hunk 6]
```diff
@@ -2,11 +2,33 @@
...
+
+Then, when we go through the list, we either milk the highest-producing cow or rent the lowest producing cow, based on which one makes more money.
```
Comment (ryanchou-dev) [inline solutions/silver/usaco-787.mdx:30]:
```suggestion
Then, when we go through the list, we'll either milk the highest-producing cow or rent the lowest-producing cow, whichever one makes more money.
```

File: solutions/silver/usaco-896.mdx (hunk 4 above)
Comment (ryanchou-dev) [reply solutions/silver/usaco-896.mdx:32]: ^^ this editorial is a tad bit too long and might confuse the reader more.

**Comment by SansPapyrus683** [comment]
here's the deal
either you do the changes or let me do them for you

**Comment by shaubby** [comment]
Sorry I've been gotten really busy with my schools finals. I should be done tomorrow and I can do it this weekend. If you would rather do it than you can make the changes.

**Review by SansPapyrus683 — APPROVED** [review]

**Review by ryanchou-dev — APPROVED** [review]

**Pull request merged**